refactor: abstract graph mode support into platform interface

yiz-liu opened the pull request:
## Purpose
Introduces a `support_graph_mode` method to the `Platform` interface to centralize the logic for determining if a backend supports graph execution.

This change replaces hardcoded checks for CUDA-like or XPU platforms with a single call to the new interface method. This improves modularity and simplifies adding graph mode support for future hardware backends.

## Test Plan
No further tests needed.

## Test Result
None

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [√] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [×] The test plan, such as providing test command.
- [×] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
- [ ] (Optional) Release notes update. If your change is user facing, please update the release notes draft in the [Google Doc](https://docs.google.com/document/d/1YyVqrgX4gHTtrstbq8oWUImOyPCKSGnJ7xtTpmXzlRs/edit?tab=t.0).
</details>

yiz-liu committed:
refactor: Abstract graph mode support into platform interface

Introduces a `support_graph_mode` method to the `Platform` interface to centralize the logic for determining if a backend supports graph execution.

This change replaces hardcoded checks for CUDA-like or XPU platforms with a single call to the new interface method. This improves modularity and simplifies adding graph mode support for future hardware backends.

Signed-off-by: Yizhou Liu <liu_yizhou@outlook.com>

zhuohan123 commented on vllm/platforms/xpu.py:
```diff
@@ -173,6 +173,10 @@ def check_and_update_config(cls, vllm_config: VllmConfig) -> None:
 
+    @classmethod
+    def support_graph_mode(cls) -> bool:
+        return True
```
@yiz-liu Does this make sense to you?

jikunshang replied:
make sense to me. xpu don't support graph mode for now. we will always set `cudagraph_mode` to None

yiz-liu replied:
Fixed.

yewentao256 reviewed (approved):
LGTM, thanks for the work!
Please merge from main to solve the conflicts

ProExpertProg reviewed (commented):
Great cleanup, thanks! Can we just wait for #24281 to land first (it's time sensitive) and then rebase this PR on top of that one?

yiz-liu commented on vllm/config/__init__.py:
```diff
@@ -2642,7 +2642,7 @@ def __post_init__(self):
             self.compilation_config.custom_ops.append("+rms_norm")
 
-        if current_platform.is_cuda_alike() or current_platform.is_xpu():
+        if current_platform.support_graph_mode():
```
Sure, will do, rebase and rename.

yiz-liu committed:
chore: Rename `support_graph_mode` to `support_static_graph_mode`

Renames the platform method to more accurately reflect that it checks for static graph support, such as CUDA graphs.

Updates the XPU platform to correctly report that it does not support static graphs. The runtime fallback for `cudagraph_mode` on XPU is also replaced with an assertion to enforce correct configuration.

Signed-off-by: Yizhou Liu <liu_yizhou@outlook.com>
